### DISCUSSION TITLE
Creating account handles using existing WalletConnect session objects, or an alternative

### DISCUSSION THREAD

**Discussion by rithik-raja** [reach-sh/reach-lang#1038] [Q&A] [answered]
In the Reach docs under "using account handles" I've noticed a lack of functionality that is otherwise provided by the conventional WalletConnect object. This includes things like sending generic transactions and disconnecting the wallet.

Please refer to the PR by xarmian, which outlines the same:
https://github.com/reach-sh/reach-lang/pull/765

Is there any alternative for this as of now? And the chances of official support?

**Comment by jeapostrophe** [comment, marked as answer]
PR #765 is about providing a custom WC... that is, it is about the PRODUCTION of an account handle in some sense.

You seem to be asking about the USE of an account handle. I think that the answer to your question is that you can use `acc.networkAccount` to get the underlying handle --- https://docs.reach.sh/frontend/#js_networkAccount --- and just do stuff that is customized to your network. You can also directly communicate with the provider --- https://docs.reach.sh/frontend/#js_setProvider --- and send arbitrary transactions.

Maybe you can help me understand what you're looking for exactly and we turn those into feature requests?

**Reply by rithik-raja** [reply]
This makes sense! I didn't realize I could use ```networkAccount``` to get this functionality. If I am correct, I can store this object and use it to restore a session via ```stdlib.connectAccount()```.

However, I currently have everything I need working perfectly fine by using PR #765. It was really easy for me to port my existing code containing Algorand WC semantics to Reach by only making a couple modifications (My code is structured as per https://developer.algorand.org/docs/get-details/walletconnect/?from_query=walletconnect#create-publication-overlay). I'm not sure if it's worth modifying all my code to use the ```networkAccount``` method. I feel that in itself is a strong reason why something similar to the PR should get official support. It would allow developers with existing WC-based dApps to adopt Reach without having to refactor their code too much.

In addition, the official WC object provides for an ```connector.on("disconnect")``` and ```connector.killsession()```  method for which I am unable to find an equivalent in the Reach docs.

Please let me know what you make of all this

**Reply by jeapostrophe** [reply]
I'll work on merging that PR and supporting a disconnect or kill options

**Reply by rithik-raja** [reply]
Thank you! I'm looking forward to it.

**Reply by jeapostrophe** [reply]
I've finally implemented this. You may enjoy checking out this example program: https://github.com/reach-sh/reach-lang/tree/master/examples/algo-wallet-demo
